tests/test_aquacomputer.py

```diff
@@ -0,0 +1,135 @@
+    if has_hwmon:
+        mockD5NextDevice._hwmon = HwmonDevice(None, None)
+
+    got = mockD5NextDevice.get_status(direct_access=direct_access)
+
+    expected = [
+        ("Liquid temperature", pytest.approx(25.1, 0.1), "°C"),
+        ("Pump speed", 1976, "rpm"),
+        ("Pump power", pytest.approx(2.58, 0.1), "W"),
+        ("Pump voltage", pytest.approx(12.02, 0.1), "V"),
+        ("Pump current", pytest.approx(0.21, 0.1), "A"),
+        ("Fan speed", 365, "rpm"),
+        ("Fan power", pytest.approx(0.38, 0.1), "W"),
+        ("Fan voltage", pytest.approx(12.04, 0.1), "V"),
+        ("Fan current", pytest.approx(0.03, 0.1), "A"),
+        ("+5V voltage", pytest.approx(5.00, 0.1), "V"),
+        ("+12V voltage", pytest.approx(12.04, 0.1), "V"),
+    ]
+
+    assert sorted(got) == sorted(expected)
+
+
+def test_d5next_get_status_from_hwmon(mockD5NextDevice, tmp_path):
+
+    mockD5NextDevice._hwmon = HwmonDevice("mock_module", tmp_path)
+    (tmp_path / "temp1_input").write_text("25100\n")  # Liquid temperature
+    (tmp_path / "fan1_input").write_text("1976\n")  # Pump speed
+    (tmp_path / "power1_input").write_text("2580000\n")  # Pump power
+    (tmp_path / "in0_input").write_text("12020\n")  # Pump voltage
+    (tmp_path / "curr1_input").write_text("215\n")  # Pump current
+    (tmp_path / "fan2_input").write_text("365\n")  # Fan speed
+    (tmp_path / "power2_input").write_text("380000\n")  # Fan power
+    (tmp_path / "in1_input").write_text("12040\n")  # Fan voltage
+    (tmp_path / "curr2_input").write_text("31\n")  # Fan current
+    (tmp_path / "in2_input").write_text("4990\n")  # +5V voltage
+    (tmp_path / "in3_input").write_text("12040\n")  # +12V voltage
+
+    got = mockD5NextDevice.get_status()
+
+    expected = [
+        ("Liquid temperature", pytest.approx(25.1, 0.1), "°C"),
+        ("Pump speed", 1976, "rpm"),
+        ("Pump power", pytest.approx(2.58, 0.1), "W"),
+        ("Pump voltage", pytest.approx(12.02, 0.1), "V"),
+        ("Pump current", pytest.approx(0.21, 0.1), "A"),
+        ("Fan speed", 365, "rpm"),
+        ("Fan power", pytest.approx(0.38, 0.1), "W"),
+        ("Fan voltage", pytest.approx(12.04, 0.1), "V"),
+        ("Fan current", pytest.approx(0.03, 0.1), "A"),
+        ("+5V voltage", pytest.approx(5.00, 0.1), "V"),
+        ("+12V voltage", pytest.approx(12.04, 0.1), "V"),
+    ]
+
+    assert sorted(got) == sorted(expected)
+
+
+def test_d5next_set_fixed_speeds_not_supported(mockD5NextDevice):
+    with pytest.raises(NotSupportedByDriver):
+        mockD5NextDevice.set_fixed_speed("fan", 42)
+
+    with pytest.raises(NotSupportedByDriver):
+        mockD5NextDevice.set_fixed_speed("pump", 84)
+
+
+def test_kraken_speed_profiles_not_supported(mockD5NextDevice):
+    with pytest.raises(NotSupportedByDriver):
+        mockD5NextDevice.set_speed_profile("fan", 1)
+
+    with pytest.raises(NotSupportedByDriver):
+        mockD5NextDevice.set_speed_profile("pump", 1)
```

Comment:
A `None` or empty list/tuple are a bit less unexpected than an int as placeholders for the profile parameter.  It doesn't matter a lot, but personally I did notice the weird int and stopped for a second to check what was going on.